base/applications/taskmgr/graphctl.c

```diff
@@ -12,666 +12,363 @@
-        /*  which now needs to be cleaned up prior to adding the new point */
-        rectCleanUp = this->m_rectPlot;
-        rectCleanUp.left  = rectCleanUp.right - this->m_nShiftPixels;
-
-        /*  fill the cleanup area with the background */
-        FillRect(this->m_dcPlot, &rectCleanUp, this->m_brushBack);
-
-        /*  draw the next line segment */
-        for (i = 0; i < MAX_PLOTS; i++)
+        SelectObject(hdcg, inst->hPen1);		
+        t = inst->PointBuffer + inst->NumberOfPoints;
+        x = inst->BitmapWidth - 1;
+        j = inst->CurrIndex;
+        y = h - *(t + j) * coef;
+        MoveToEx(hdcg, x, y, NULL);
+        for (i = 0; i < inst->NumberOfPoints; i++)
         {
-            /*  grab the plotting pen */
-            oldPen = (HPEN)SelectObject(this->m_dcPlot, this->m_penPlot[i]);
-
-            /*  move to the previous point */
-            prevX = this->m_rectPlot.right-this->m_nPlotShiftPixels;
-            prevY = this->m_rectPlot.bottom -
-                (long)((this->m_dPreviousPosition[i] - this->m_dLowerLimit) * this->m_dVerticalFactor);
-            MoveToEx(this->m_dcPlot, prevX, prevY, NULL);
-
-            /*  draw to the current point */
-            currX = this->m_rectPlot.right-this->m_nHalfShiftPixels;
-            currY = this->m_rectPlot.bottom -
-                (long)((this->m_dCurrentPosition[i] - this->m_dLowerLimit) * this->m_dVerticalFactor);
-            LineTo(this->m_dcPlot, currX, currY);
-
-            /*  Restore the pen  */
-            SelectObject(this->m_dcPlot, oldPen);
-
-            /*  if the data leaks over the upper or lower plot boundaries
-             *  fill the upper and lower leakage with the background
-             *  this will facilitate clipping on an as needed basis
-             *  as opposed to always calling IntersectClipRect
-             */
-            if ((prevY <= this->m_rectPlot.top) || (currY <= this->m_rectPlot.top))
-            {
-                RECT rc;
-                rc.bottom = this->m_rectPlot.top + 1;
-                rc.left = prevX;
-                rc.right = currX + 1;
-                rc.top = this->m_rectClient.top;
-                FillRect(this->m_dcPlot, &rc, this->m_brushBack);
-            }
-            if ((prevY >= this->m_rectPlot.bottom) || (currY >= this->m_rectPlot.bottom))
-            {
-                RECT rc;
-                rc.bottom = this->m_rectClient.bottom + 1;
-                rc.left = prevX;
-                rc.right = currX + 1;
-                rc.top = this->m_rectPlot.bottom + 1;
-                /* RECT rc(prevX, m_rectPlot.bottom+1, currX+1, m_rectClient.bottom+1); */
-                FillRect(this->m_dcPlot, &rc, this->m_brushBack);
-            }
-
-            /*  store the current point for connection to the next point */
-            this->m_dPreviousPosition[i] = this->m_dCurrentPosition[i];
+            if(!j)  j = inst->NumberOfPoints - 1;
+            else    j--;
```

Comment:
```suggestion
            j = (j ? j : inst->NumberOfPoints) - 1;
```

Reply:
your suggestion doesn't match. on `j != 0`, it should decrement.

Reply:
I copy-pasted the same code you wrote below (line 237), which does exactly what your if-else do:
- if `j != 0` originally (the `else` clause), then the ?: operation returns the old j value, which is then decreased by one, and then stored into `j`;
- if `j == 0` (the first case), then the ?: operation returns `inst->NumberOfPoints`, whose value is decreased by one and stored into `j`.

Reply:
right, this is how more "readable" your formatting is - I misread it. it's definitely less clear, than the original. seriously, @HBelusca why are you doing this? what for? what is wrong with the original here?

Reply:
> @HBelusca why are you doing this? what for? what is wrong with the original here?

As I already mentioned in the previous comment:
> I copy-pasted the same code you wrote below (line 237)

i.e. I merely pointed what was already written elsewhere in the PR before I came commenting there.

Reply:
I wrote it in both places the same, like this:
`if(!j)  j = inst->NumberOfPoints - 1;`
`else j--;`
it's figuring the needed index on a circular buffer, and how I did it originally, looked much more clear in that respect, than that almost brainf&ck like expression. but the point "why" doing these changes remain unclear. feels like if I wrote with the `?:` operator, you or another style warrior, would change it to `if else`. guys, please, just stop it, it's merely nitpicking, it's not even the styling, just changes for the sake of changes.

Reply:
Yeah it was my code suggestion actually. Just because 1 line is shorter than 4 (which it would be if we're about to save original style).